File: app/src/main/java/com/example/kaola/myapplication/designpattern/decorator/java/factory/FruitDumplings.java
package com.example.kaola.myapplication.designpattern.decorator.java.factory;

/**
 * @author zhangchao on 2019-07-18.
 */

public class FruitDumplings extends ADumplings {
    Dumplings dumplings;

    @Override
    public void makeDumplings() {
        dumplings = new Dumplings();
        dumplings.setName("我是水果饺子");
    }

    @Override
    public Dumplings getDumpling() {
        return dumplings;
    }
}

File: app/src/main/java/com/example/kaola/myapplication/designpattern/decorator/kotlin/factory/FruitDumplings.kt
package com.example.kaola.myapplication.designpattern.decorator.kotlin.factory

import android.util.Log


/**
 * @author zhangchao on 2019-07-18.
 */

class FruitDumplings : ADumplings() {
    lateinit var aDumplings: Dumplings

    override fun makeDumplings() {
        aDumplings = Dumplings("水果")
    }

    override fun getDumplings(): Dumplings {
        Log.d("logx", "返回kotlin类型的水果饺子")
        return aDumplings
    }

}